modules/smartattributes/lib/Auth/Process/SmartID.php

```diff
@@ -1,116 +1,123 @@
-			if (!is_array($this->_candidates)) {
-				throw new Exception('SmartID authproc configuration error: \'candidates\' should be an array.');
-			}
-		}
-
-		if (array_key_exists('id_attribute', $config)) {
-			$this->_id_attribute = $config['id_attribute'];
-			if (!is_string($this->_id_attribute)) {
-				throw new Exception('SmartID authproc configuration error: \'id_attribute\' should be a string.');
-			}
-		}
-
-		if (array_key_exists('add_authority', $config)) {
-			$this->_add_authority = $config['add_authority'];
-			if (!is_bool($this->_add_authority)) {
-				throw new Exception('SmartID authproc configuration error: \'add_authority\' should be a boolean.');
-			}
-		}
-
-		if (array_key_exists('add_candidate', $config)) {
-			$this->_add_candidate = $config['add_candidate'];
-			if (!is_bool($this->_add_candidate)) {
-				throw new Exception('SmartID authproc configuration error: \'add_candidate\' should be a boolean.');
-			}
-		}
-
-	}
-
-	private function addID($attributes, $request) {
-		foreach ($this->_candidates as $idCandidate) {
-			if (isset($attributes[$idCandidate][0])) {
-				if(($this->_add_authority) && (isset($request['saml:AuthenticatingAuthority'][0]))) {
-					return ($this->_add_candidate ? $idCandidate.':' : '').$attributes[$idCandidate][0] . '!' . $request['saml:AuthenticatingAuthority'][0];
-				} else {
-					return ($this->_add_candidate ? $idCandidate.':' : '').$attributes[$idCandidate][0];
-				}
-			}
-		}
-		/*
-		* At this stage no usable id_candidate has been detected.
-		*/
-		throw new SimpleSAML_Error_Exception('This service needs at least one of the following
-		attributes to identity users: '.implode(', ', $this->_candidates).'. Unfortunately not
-		one of them was detected. Please ask your institution administrator to release one of
-		them, or try using another identity provider.');
-	}
-
-
-	/**
-	 * Apply filter to add or replace attributes.
-	 *
-	 * Add or replace existing attributes with the configured values.
-	 *
-	 * @param array &$request  The current request
-	 */
-	public function process(&$request) {
-		assert(is_array($request));
-		assert(array_key_exists('Attributes', $request));
-
-		$ID = $this->addID($request['Attributes'], $request);
-
-		if(isset($ID)) $request['Attributes'][$this->_id_attribute] = array($ID);
-	}
+class sspmod_smartattributes_Auth_Process_SmartID extends SimpleSAML_Auth_ProcessingFilter
+{
+    /**
+     * Which attributes to use as identifiers?
+     *
+     * IMPORTANT: If you use the (default) attributemaps (twitter2name, facebook2name,
+     * etc., be sure to comment out the entries that map xxx_targetedID to
+     * eduPersonTargetedID, or there will be no way to see its origin any more.
+     */
+    private $_candidates = array(
+        'eduPersonTargetedID',
+        'eduPersonPrincipalName',
+        'openid',
+        'facebook_targetedID',
+        'twitter_targetedID',
+        'windowslive_targetedID',
+        'myspace_targetedID',
```

Comment:
Is this related?

Reply:
It's just whitespace fixes you're looking at.. This code is not new.

Reply:
No, I was skipping whitespace changes explicitly, and then the `myspace_targetedID` appeared as new. That's why I was wondering whether it was related to the bugfix or not... 🤔

Reply:
I guess a merge error since the line has been removed in 852ee7e32a75aac93263e6b465567eedfd0f02d5.

Reply:
In that case, I think it's a good idea to remove any leftovers. Other than that: 👍

Reply:
I'll fix it later today